src/kibana/plugins/settings/sections/advanced/index.js

```diff
@@ -10,59 +10,49 @@ define(function (require) {
   });
 
   require('modules').get('apps/settings')
-  .directive('kbnSettingsAdvanced', function (config, Notifier, Private) {
+  .directive('kbnSettingsAdvanced', function (config, Notifier, Private, $rootScope) {
     return {
       restrict: 'E',
       link: function ($scope) {
-        var notify = new Notifier();
-        var configVals = config._vals();
         var keyCodes = {
           ESC: 27
         };
 
-        // determine if a value is too complex to be edditted (at this time)
-        var tooComplex = function (conf) {
-          // get the type of the current value or the default
-          switch (conf.type) {
-          case 'string':
-          case 'number':
-          case 'null':
-          case 'undefined':
-            conf.normal = true;
-            break;
-          case 'json':
-            conf.json = true;
-            break;
-          default:
-            if (_.isArray(config.get(conf.name))) {
-              conf.array = true;
-            } else if (typeof config.get(conf.name) === 'boolean') {
-              conf.bool = true;
-            } else {
-              conf.tooComplex = true;
-            }
-          }
-        };
+        function getEditorType(conf) {
+          if (_.contains('number string null undefined', conf.type)) return 'normal';
+          if (_.contains('json array boolean', conf.type)) return conf.type;
```

Comment:
Not a huge deal here, but this will do partial string match, instead of a word match, so you may not always get the result you want when checking `conf.type`